### PR TITLE
Per-game instead of per-group EGD exporting.

### PR DESCRIPTION
**Pull request by lukaszlew**
Depends on PR #56 

 # EGD Export Feature Improvements

  This branch adds enhanced functionality for exporting Go tournament data to the European Go Database (EGD) format with comprehensive testing and error handling.

  Key Changes

  1. Per-Game EGD Eligibility
    - Added is_egd_eligible property to Game model
    - Games are eligible if both players have approved EGD reporting
    - Visual UI indicators for EGD-eligible games (blue border, EGD badge)
  2. Improved Error Handling
    - Added validation for missing ranks and EGD PINs
    - Clear error messages identify which players need information
    - Includes links to EGD website for PIN lookup
  3. Enhanced UI Elements
    - Added EGD status information on game detail page
    - Show globe icon for players who have agreed to EGD reporting
    - Added EGD badge to eligible games in lists
  4. Comprehensive Tests
    - Test various edge cases (BYE games, not played games, forfeited games)
    - Test export generation with rounds having no eligible games
    - Test eligibility criteria and export format
  5. Documentation Changes
    - Updated help text to clarify Group.is_egd is deprecated
    - Added info tooltips to explain EGD reporting

  Code Reading Order

  1. Models & Core Logic
    - league/models.py - Game.is_egd_eligible property
    - league/utils/egd.py - EGD export formatting functions
  2. Views & Export Logic
    - league/views.py - GroupEGDExportView with validation
  3. Tests
    - league/tests/utils/test_egd.py - Unit tests for export formatting
    - league/tests/test_egd_export.py - Integration tests for eligibility and export
  4. UI Elements
    - templates/league/group_detail.html - EGD badges and info
    - templates/league/game_detail.html - Detailed EGD status section
    - templates/league/includes/player_badge.html - EGD approval icon
  5. Style Changes
    - static/main.css - Styles for EGD indicators

  Technical Notes

  - Group.is_egd is now considered deprecated but kept for backward compatibility
  - Export now works on a per-game basis rather than per-group
  - Export file contains clear error messages if data is missing
  - Improved organization of games by round in export format

### CODE REVIEW

File: iglo/league/models.py [hunk 1]
```diff
@@ -282,18 +284,20 @@ def _assign_players_to_groups(self, players: list["Player"]) -> None:
...
         for group_order in range(math.ceil(len(players) / self.players_per_group)):
             group_players = players[group_order * self.players_per_group : (group_order + 1) * self.players_per_group]
+            # Note: is_egd is deprecated, we now check per-game eligibility
+            # but we keep this for backward compatibility
             is_egd = all(p.egd_approval for p in group_players)
             if len(group_players) < max((self.players_per_group - 1), 2) and last_group:
                 group = last_group
                 group.type = GroupType.MCMAHON
-                group.is_egd = group.is_egd and is_egd
+                group.is_egd = group.is_egd and is_egd  # Deprecated - games are now eligible based on individual player approvals
```
Comment (CezCz) [inline iglo/league/models.py:293]: we don't need to repeat the deprecation notice everywhere it should be done in the entity class. (group.is_egd should be deprecated)

File: iglo/league/models.py [hunk 2]
```diff
@@ -1019,6 +1023,17 @@ def is_delayed(self):
...
+    @property
+    def is_egd_eligible(self) -> bool:
+        """
+        Determine if a game is eligible for EGD/EGF submission based on player approvals.
+        Both players must exist (not BYE games) and have egd_approval.
+        The game doesn't need to be played yet to be marked as eligible in the UI.
+        """
+        return (self.black and self.white and 
+                self.black.player.egd_approval and 
+                self.white.player.egd_approval)
```
Comment (CezCz) [inline iglo/league/models.py:1036]: what if a player plays a couple of games and changes their mind?

File: iglo/league/tests/test_egd_export.py [hunk 3]
```diff
@@ -0,0 +1,466 @@
...
+        
+        self.game8 = GameFactory(
+            group=self.group,
+            round=self.round4,
+            black=self.member4,
+            white=self.member2,
+            win_type=None  # Not played
+        )
+
+    def test_complex_egd_export_with_edge_cases(self):
```
Comment (CezCz) [inline iglo/league/tests/test_egd_export.py:238]: instead of writing "with_edge_cases" I'd just write what edge case and test single edge case per test

File: iglo/league/models.py (hunk 1 above)
Comment (lukaszlew) [reply iglo/league/models.py:293]: Done.

File: iglo/league/models.py (hunk 2 above)
Comment (lukaszlew) [reply iglo/league/models.py]: Thanks for catching it.
Fixed ... I believe.

File: iglo/league/tests/test_egd_export.py (hunk 3 above)
Comment (lukaszlew) [reply iglo/league/tests/test_egd_export.py:238]: I forgot to uncomment the assertion here. 
Fixed.

But I'm not willing to work more on this test, its annoying. 
I can delete it, but it serves a value when running all tests.

**Review by lukaszlew — COMMENTED** [review]
Answered.

File: iglo/league/models.py [hunk 4]
```diff
@@ -188,18 +188,14 @@ def create_groups(self, use_igor: bool, pairing_type: str=PairingType.DEFAULT, b
...
-                type=GroupType.BANDED,  # Using our new BANDED type
-                is_egd=is_egd,
+                type=GroupType.BANDED,
+                is_egd=False,
+                band_size=band_size,
+                point_difference=point_difference,
             )
-
-            # Store the band_size and point_difference for later use
-            group.band_size = band_size
```
Comment (CezCz) [inline iglo/league/models.py:201]: why do we delete this?

File: iglo/league/models.py (hunk 1 above)
Comment (CezCz) [reply iglo/league/models.py:274]: the comments are still there

File: iglo/league/models.py [hunk 5]
```diff
@@ -859,7 +862,7 @@ def create_bye_game(self, group: Group, round: Round, member: Member):
     def get_immediate_games(self):
         now = datetime.datetime.now()
         start = now + datetime.timedelta(hours=2)
-        end = now + datetime.timedelta(days=8)
+        end = now + datetime.timedelta(days=6, hours=23)
```
Comment (CezCz) [inline iglo/league/models.py:865]: why is this changed in that pr? It seems there's some divergence between branches fixed in this pr

File: iglo/league/models.py (hunk 4 above)
Comment (lukaszlew) [reply iglo/league/models.py:201]: Line 196 should do the same, no?

File: iglo/league/models.py (hunk 5 above)
Comment (lukaszlew) [reply iglo/league/models.py]: It was devel->egd merge from old devel I think. 
Is it fixed?

**Review by lukaszlew — COMMENTED** [review]
PTAL
